core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -783,13 +826,22 @@ public void requestContent(@Nullable Object requestContent, @Nullable Object raw
             return;
         }
 
+        int requestCompletionFlags = RequestLogProperty.FLAGS_REQUEST_COMPLETE & ~deferredFlags;
+        if (needToDeferName() && name == null) {
+            requestCompletionFlags &= ~RequestLogProperty.NAME.flag();
+        }
+
         this.requestContent = requestContent;
         this.rawRequestContent = rawRequestContent;
         updateFlags(RequestLogProperty.REQUEST_CONTENT);
 
         if (requestContent instanceof RpcRequest && ctx.rpcRequest() == null) {
             ctx.updateRpcRequest((RpcRequest) requestContent);
         }
+
+        if (isAvailable(requestCompletionFlags)) {
+            setNamesIfAbsent();
```

Comment:
Can't we just call `setNamesIfAbsent()` if `name == null`?

Reply:
`setNamesIfAbsent()` make `NAME` available. You cannot set names with `logBuilder.name(...)` even before `REQUEST_COMPLETE`.
 I intended to give a constant chance of setting names to users.